src/main/java/net/backupcup/mcde/screen/handler/GildingFoundryScreenHandler.java

```diff
@@ -1,28 +1,53 @@
 package net.backupcup.mcde.screen.handler;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import net.backupcup.mcde.MCDEnchantments;
 import net.backupcup.mcde.block.ModBlocks;
+import net.backupcup.mcde.block.entity.GildingFoundryBlockEntity;
 import net.backupcup.mcde.util.EnchantmentSlots;
 import net.backupcup.mcde.util.EnchantmentUtils;
+import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
+import net.fabricmc.fabric.api.networking.v1.PacketSender;
+import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.entity.player.PlayerInventory;
 import net.minecraft.inventory.Inventory;
 import net.minecraft.inventory.SimpleInventory;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
+import net.minecraft.network.PacketByteBuf;
 import net.minecraft.screen.ArrayPropertyDelegate;
 import net.minecraft.screen.PropertyDelegate;
 import net.minecraft.screen.ScreenHandler;
 import net.minecraft.screen.ScreenHandlerContext;
+import net.minecraft.screen.ScreenHandlerListener;
 import net.minecraft.screen.slot.Slot;
+import net.minecraft.util.Identifier;
 
-public class GildingFoundryScreenHandler extends ScreenHandler {
+public class GildingFoundryScreenHandler extends ScreenHandler implements ScreenHandlerListener {
+    public static final Identifier GILDING_PACKET = Identifier.of(MCDEnchantments.MOD_ID, "gilding");
     private final Inventory inventory;
+    private final PlayerEntity player;
```

Comment:
As mentioned last time:
Yarn uses `playerEntity` rather than `player`. As this can cause confusion for the reader knowing which mappings are being used, consider the aforementioned suggestion.